sympy/simplify/cse_main.py

```diff
@@ -212,42 +212,120 @@ def _find_opts(expr):
 
     ## Process Adds and commutative Muls
 
-    def _match_common_args(Func, funcs):
+    def _match_common_args(Func, funcs, dict, F):
         if order != 'none':
             funcs = list(ordered(funcs))
         else:
             funcs = sorted(funcs, key=lambda x: len(x.args))
 
-        func_args = [set(e.args) for e in funcs]
+        # ----------------- helpers ---------------------------
+        def cfunc(dx, dy, c):
```

Comment:
Is it possible to move these functions out to the module level? They are impossible to test independently if they are inside another function.